src/rgw/rgw_rest_s3.cc

```diff
@@ -4995,7 +4995,7 @@ int RGWHandler_REST_S3::init(rgw::sal::Driver* driver, req_state *s,
   s->has_acl_header = s->info.env->exists_prefix("HTTP_X_AMZ_GRANT");
 
   const char *copy_source = s->info.env->get("HTTP_X_AMZ_COPY_SOURCE");
-  if (copy_source &&
+  if (copy_source && copy_source[0] != '\0' &&
```

Comment:
is this change necessary? if the client sends an empty `x-amz-copy-source` header, this function should still treat it as a copy request and catch the error from `parse_copy_location()` just below. but your change will cause us to skip this block and continue on with a null `s->src_object`. that might cause us to crash or misbehave later in the request processing